IntegrationTests/Services/AWSSTSIntegrationTests/STSWebIdentityCredentialsProviderTests.swift

```diff
@@ -0,0 +1,227 @@
+                identityPoolName: identityPoolName
+        ))
+        identityPoolId = identityPool.identityPoolId
+    }
+
+    private func saveTokenIntoFile() throws {
+        oidcTokenFilePath = FileManager.default.temporaryDirectory.appendingPathComponent("token.txt").pathExtension
+        let tokenData = oidcToken.data(using: String.Encoding.utf8)
+        let fileCreated = FileManager.default.createFile(
+            atPath: oidcTokenFilePath, contents: tokenData, attributes: nil
+        )
+        if !fileCreated {
+            throw TokenFileError.TokenFileCreationFailed("Failed to create token text file.")
+        }
+    }
+
+    private func getAndCacheOIDCTokenFromCognito() async throws {
+        // Get Cognito ID from Cognito identity pool
+        stsClient = try STSClient(region: region)
+        let accountId = try await stsClient.getCallerIdentity(input: GetCallerIdentityInput()).account
+        let cognitoId = try await cognitoIdentityClient.getId(input: GetIdInput(
+            accountId: accountId, identityPoolId: identityPoolId
+        ))
+        // Get OIDC token from Cognito identity pool using Cognito ID
+        oidcToken  = try await cognitoIdentityClient.getOpenIdToken(
+            input: GetOpenIdTokenInput(identityId: cognitoId.identityId)
+        ).token
+        // Save OIDC token to a file then save filepath
+        try saveTokenIntoFile()
+    }
+
+    private func constructSTSConfigWithWebIdentityCredentialsProvider() async throws {
+        let webIdentityCredentialsProvider = try STSWebIdentityCredentialsProvider(
+            region: region,
+            roleArn: roleArn,
+            roleSessionName: roleSessionName,
+            tokenFilePath: oidcTokenFilePath
+        )
+        stsConfig = try await STSClient.STSClientConfiguration(
+            credentialsProvider: webIdentityCredentialsProvider,
+            region: region
+        )
+    }
+
+    /* /////////////////////////
+     * TEARDOWN HELPER FUNCTIONS
+     * /////////////////////////
+     */
+
+    private func deleteInlineRolePolicy() async throws {
+        let policyName = try await iamClient.listRolePolicies(input: ListRolePoliciesInput(roleName: roleName)).policyNames
+        _ = try await iamClient.deleteRolePolicy(input: DeleteRolePolicyInput(
+            policyName: policyName?[0], roleName: roleName
+        ))
+    }
+
+    private func deleteTokenFile() throws {
+        do {
+            try FileManager.default.removeItem(atPath: oidcTokenFilePath)
+        } catch {
+            throw TokenFileError.TokenFileDeletionFailed("Failed to delete token text file.")
+        }
+    }
+
+    /* //////////////////////////////
+     * ENUMS USED IN SETUP & TEARDOWN
+     * //////////////////////////////
+     */
+
+    enum TokenFileError: Error {
+        case TokenFileCreationFailed(String)
+        case TokenFileDeletionFailed(String)
+    }
+
+    enum IdentityPoolStatus {
```

Comment:
Swift convention is that enum cases should be lowerCamelCase.

Reply:
Changed as suggested.